lua/nvim-treesitter/locals.lua

```diff
@@ -89,4 +106,39 @@ function M.get_references(bufnr)
   return refs
 end
 
+
+--- Return all nodes in locals corresponding to a specific capture (like @scope, @reference)
+-- Works like M.get_references or M.get_scopes except you can choose the capture
+-- Can also be a nested capture like @definition.function to get all nodes defining a function
+function M.get_capture_matches(bufnr, capture_string, locals_kind)
+    if not string.sub(capture_string, 1,2) == '@' then
+      print('capture_string must start with "@"')
+      return
+    end
+
+    -- Build capture path: @definition.function -> ["definition", "function"]
+    local capture_path = vim.split(string.sub(capture_string, 2), '%.')
+
+    local matches = {}
+    -- For each match
+    for _, match in pairs(M.get_locals(bufnr, locals_kind)) do
+      local insert = match
+      -- Follow capture path while we still have a table
+      for _, capture_segment in pairs(capture_path) do
+        if type(insert) == 'table' then
+          insert = insert[capture_segment]
+        end
+      end
+
+      -- Following the capture path was successful
```

Comment:
i think you can remove the comments in this function, the code is already pretty explicit :)

Reply:
I hope I can use the function from the highlight-definition PR which is more general than this one.